src/solvers/base_solver.cpp

```diff
@@ -4,193 +4,183 @@
               "State and coefficient bundles not the same size in BaseSolver::ProjectState.");
 
-  for (int i=0; i<state_coef.Size(); ++i) {
-    m_state[i].gf->ProjectCoefficient(*state_coef[i]);
+  for (size_t i = 0; i < state_coef.size(); ++i)
+  {
+    m_states[i].gf->ProjectCoefficient(*state_coef[i]);
   }
 }
 
-void BaseSolver::ProjectState(mfem::Array<mfem::VectorCoefficient*> state_vec_coef)
+void BaseSolver::ProjectState(std::vector<mfem::VectorCoefficient *> state_vec_coef)
 {
-  MFEM_ASSERT(state_vec_coef.Size() == m_state.size(),
+  MFEM_ASSERT(state_vec_coef.size() == m_states.size(),
               "State and coefficient bundles not the same size in BaseSolver::ProjectState.");
 
-  for (int i=0; i<state_vec_coef.Size(); ++i) {
-    m_state[i].gf->ProjectCoefficient(*state_vec_coef[i]);
+  for (size_t i = 0; i < state_vec_coef.size(); ++i)
+  {
+    m_states[i].gf->ProjectCoefficient(*state_vec_coef[i]);
   }
 }
 
 void BaseSolver::SetState(const std::vector<FiniteElementState> &state)
 {
   MFEM_ASSERT(state.size() > 0, "State vector array of size 0 in BaseSolver::SetState.");
-  m_state = state;
+  m_states = state;
 
-  MPI_Comm_rank(m_state.begin()->space->GetComm(), &m_rank);
+  MPI_Comm_rank(m_states[0].space->GetComm(), &m_rank);
 }
 
-std::vector< FiniteElementState > BaseSolver::GetState() const
-{
-  return m_state;
-}
+std::vector<FiniteElementState> BaseSolver::GetState() const { return m_states; }
 
 void BaseSolver::SetTimestepper(const TimestepMethod timestepper)
 {
   m_timestepper = timestepper;
 
-  switch (m_timestepper) {
-  case TimestepMethod::QuasiStatic :
-    m_ode_solver = nullptr;
-    break;
-  case TimestepMethod::BackwardEuler  :
-    m_ode_solver = std::make_shared< mfem::BackwardEulerSolver >();
-    break;
-  case TimestepMethod::SDIRK33    :
-    m_ode_solver = std::make_shared< mfem::SDIRK33Solver >();
-    break;
-  case TimestepMethod::ForwardEuler :
-    m_ode_solver = std::make_shared< mfem::ForwardEulerSolver >();
-    break;
-  case TimestepMethod::RK2    :
-    m_ode_solver = std::make_shared< mfem::RK2Solver >(0.5);
-    break;
-  case TimestepMethod::RK3SSP   :
-    m_ode_solver = std::make_shared< mfem::RK3SSPSolver >();
-    break;
-  case TimestepMethod::RK4    :
-    m_ode_solver = std::make_shared< mfem::RK4Solver >();
-    break;
-  case TimestepMethod::GeneralizedAlpha :
-    m_ode_solver = std::make_shared< mfem::GeneralizedAlphaSolver >(0.5);
-    break;
-  case TimestepMethod::ImplicitMidpoint :
-    m_ode_solver = std::make_shared< mfem::ImplicitMidpointSolver >();
-    break;
-  case TimestepMethod::SDIRK23    :
-    m_ode_solver = std::make_shared< mfem::SDIRK23Solver >();
-    break;
-  case TimestepMethod::SDIRK34    :
-    m_ode_solver = std::make_shared< mfem::SDIRK34Solver >();
-    break;
-  default:
-    mfem::mfem_error("Timestep method not recognized!");
+  switch (m_timestepper)
+  {
+    case TimestepMethod::QuasiStatic: m_ode_solver = nullptr; break;
```

Comment:
its nice that the case is indented but the break got moved which is less than ideal.

Reply:
I'm generally anti multiple statements on one line.